gcloud/datastore/query.py

```diff
@@ -66,15 +69,22 @@ class Query(_implicit_environ._DatastoreBase):
     """Mapping of operator strings and their protobuf equivalents."""
 
     def __init__(self,
+                 dataset_id=None,
                  kind=None,
-                 dataset=None,
                  namespace=None,
                  ancestor=None,
                  filters=(),
                  projection=(),
                  order=(),
                  group_by=()):
-        super(Query, self).__init__(dataset=dataset)
+
+        if dataset_id is None:
```

Comment:
Yeah [PEP8][1] is somewhat explicit about this:

> Also, beware of writing `if x` when you really mean `if x is not None` -- e.g.
> when testing whether a variable or argument that defaults to `None` was
> set to some other value. The other value might have a type (such as a
> container) that could be false in a boolean context!

[1]: https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=True/False_evaluations#True/False_evaluations
[2]: https://www.python.org/dev/peps/pep-0008/